src/query/block/meta.go

```diff
@@ -73,32 +75,70 @@ func NewResultMetadata() ResultMetadata {
 	}
 }
 
-// CombineMetadata combines two result metadatas.
-func (m ResultMetadata) CombineMetadata(other ResultMetadata) ResultMetadata {
-	var combinedWarnings Warnings
-	if len(m.Warnings) == 0 {
-		if len(other.Warnings) != 0 {
-			combinedWarnings = other.Warnings
+func combineResolutions(a, b []int64) []int64 {
```

Comment:
Is Resolutions supposed to be de-duped?  i.e. if we return 10s and 10s twice, is that ok?  Not sure how this is used that much in m3query (only really know that it was used for frontend code in plugins we used to use in Grafana).

Reply:
I think you pointed out in a further comment, but Resolutions here contains a value per series that is only really used for graphite queries, specifically for remote graphite since there was an existing mechanism (replaced in this PR) for doing local queries